akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -390,11 +392,38 @@ class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Graph
    * If the split predicate `p` throws an exception and the supervision decision
    * is [[akka.stream.Supervision#resume]] or [[akka.stream.Supervision#restart]]
    * the element is dropped and the stream and substreams continue.
+   *
+   * See also [[Flow.splitAfter]].
    */
   def splitWhen(p: japi.Predicate[Out]): javadsl.Flow[In, Source[Out, Unit], Mat] =
     new Flow(delegate.splitWhen(p.test).map(_.asJava))
 
   /**
+   * This operation applies the given predicate to all incoming elements and
+   * emits them to a stream of output streams. It *ends* the current substream when the
+   * predicate is true. This means that for the following series of predicate values,
+   * three substreams will be produced with lengths 1, 2, and 3:
+   *
+   * {{{
+   * true,               // element goes into first substream
```

Comment:
Maybe, also add one or more leading `false` values here, to make it sure that they are included in the first stream.